services/scraper/search_keyword.go

```diff
@@ -0,0 +1,86 @@
+package scraper
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/gocolly/colly/v2"
+)
+
+type SearchKeywordService struct {
+	Keyword string
+
+	synchronous  bool
+	searchResult *searchKeywordResult
+}
+
+type searchKeywordResult struct {
+	Keyword  string
+	VisitURL string
+	NonAds   []string
+	OtherAds []string
+	TopAds   []string
+}
+
+var selectorList = map[string]string{
+	"nonAds":        "#search .g .yuRUbf > a",
+	"bottomLinkAds": "#tadsb .d5oMvf > a",
+	"otherAds":      "#rhs .pla-unit a.pla-unit-title-link",
+	"topImageAds":   "#tvcap .pla-unit a.pla-unit-title-link",
+	"topLinkAds":    "#tads .d5oMvf > a",
+}
+
+const collectingURLPattern = "https://www.google.com/search?q=%s&lr=lang_en"
+
+// Call handles making a service call to scrape some data from google search engine with a given keyword.
+// It will return an error when the collector can not visit the url.
+func (service *SearchKeywordService) Call() {
```

Comment:
`Call` seems very inspired by Ruby service objects 🙈  Have you seen it used in other libraries? So far, from what I have seen, `Run` would be more idiomatic in a Go application (even Beego has it). `Search` would also work.

Reply:
@olivierobert Yeah I use `Call` as we use in Ruby 🤣
I haven't check more deep with other package in Go.
For now, I change it to `Run`